File: python/stack_sorter.py
class StackSorter:
    def sort(self, stack):
        aux_stack = []

        while len(stack) > 0:
            temp = stack.pop()

            while len(aux_stack) > 0 and aux_stack[-1] > temp:
                stack.append(aux_stack.pop())
            
            aux_stack.append(temp)

        # Regresar elementos al stack original
        while len(aux_stack) > 0:
            stack.append(aux_stack.pop())

File: src/utils/StackSorter.java
package utils;

import java.util.Stack;

public class StackSorter {

    public void sort(Stack<Integer> stack) {
        Stack<Integer> auxStack = new Stack<>();

        while (!stack.isEmpty()) {
            int temp = stack.pop();

            while (!auxStack.isEmpty() && auxStack.peek() > temp) {
                stack.push(auxStack.pop());
            }

            auxStack.push(temp);
        }

        while (!auxStack.isEmpty()) {
            stack.push(auxStack.pop());
        }
    }
}